### PR TITLE
Client server add

### PR DESCRIPTION
**Pull request by BichraiX**
Added /add endpoint from the spec : POST /_matrix/client/v3/account/3pid/add
Modified UI Authentication. Since endpoints don't all have access to the same flows, I added allowedFlows as an argument to the uiauthenticate method, and defined allowedFlows as exported variable in utils/uiauth. Those variables could (should ?) be defined in the config, and need to be thought of carefully before going into production. 
Added m.login.email.identity and m.login.msisdn types for ui authentication : https://spec.matrix.org/v1.11/client-server-api/#email-based-identity--homeserver

### CODE REVIEW

File: packages/matrix-client-server/src/types.ts [hunk 1]
```diff
@@ -193,6 +191,11 @@ interface ApplicationServiceAuth {
   username: string
 }
 
+interface SsoAuth {
```
Comment (guimard) [inline packages/matrix-client-server/src/types.ts:194]: Looks to be a duplication, see line 168

File: packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts [hunk 2]
```diff
@@ -6,29 +6,129 @@ import {
...
+              name: 'Terms of Service',
+              url: 'https://example.org/somewhere/terms-1.2-en.html'
+            },
+            fr: {
+              name: "Conditions d'utilisation",
+              url: 'https://example.org/somewhere/terms-1.2-fr.html'
+            }
+          }
+        }
+      }
```
Comment (guimard) [inline packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts:47]: Temporary fixed value ?

Comment (BichraiX) [reply packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts:47]: Yes, I believe these should be included in the config in the future but I've just included this from an example of the spec for testing purposes. I'll add a comment to clarify this

Comment (guimard) [reply packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts:49]: Then add a TODO

Comment (BichraiX) [reply packages/matrix-client-server/src/utils/userInteractiveAuthentication.ts:49]: Done

**Pull request merged**